src/components/survey/Survey.js

```diff
@@ -63,6 +63,7 @@ export class SurveyComponent extends BaseComponent {
 
   setValue(value, flags) {
     flags = this.getFlags.apply(this, arguments);
+    this.data[this.component.key] = value;
```

Comment:
Shouldn't this be within the ```if (!value)``` statement? I suppose you needed to add this in case it is a "reset" of value and it does not get into ```updateValue```?